server/pulp/server/async/tasks.py

```diff
@@ -409,12 +409,21 @@ def cancel(task_id):
     task_status = TaskStatusManager.find_by_task_id(task_id)
     if task_status is None:
         raise MissingResource(task_id)
+
     if task_status['state'] in constants.CALL_COMPLETE_STATES:
         # If the task is already done, just stop
         msg = _('Task [%(task_id)s] already in a completed state: %(state)s')
         logger.info(msg % {'task_id': task_id, 'state': task_status['state']})
         return
-    controller.revoke(task_id, terminate=True)
+
+    if task_status.get('worker_name') == 'agent':
+        tag_dict = dict([tags.parse_resource_tag(t) for t in task_status['tags']])
```

Comment:
Might be nice if the *tags* module had a function to convert the task's tags into a dict but did not feel like adding it.